app/scripts/directives/document-line-edit.js

```diff
@@ -149,13 +151,21 @@
           return scope.numLines > 1;
         };
 
+        scope.warningMessages = function(i) {
+          return messageFor('Warning', i);
+        };
+
+        scope.errorMessages = function(i) {
+          return messageFor('Error', i);
+        };
+
         scope.getHelpTextForKey = function() {
           var helpText = '';
           if (!lodash.isEmpty($rootScope.keysHelpText)) {
             var node = lodash.find($rootScope.keysHelpText, function(l) {
               return (scope.line.name in l);
             });
-            helpText = node ? node[scope.line.name] : 'Key is invalid.';
+            helpText = node ? node[scope.line.name] : '';
```

Comment:
@rupakg - I'm wondering if we can get rid of all the client side key checking logic here... That may be a good technical debt chore for you to bite off.